### ISSUE TITLE
`Lint/RedundantCopDisableDirective` leaves an orphan `enable` directive when the cop is not enabled

### ISSUE THREAD

**Issue by sanfrecce-osaka** [rubocop/rubocop#15671]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Lint/RedundantCopDisableDirective, Lint/RedundantCopEnableDirective

### Steps to reproduce the problem

This happens with a matching `disable` / `enable` pair when the target cop is not enabled.

1. Create a `.rubocop.yml` with the following content:

```yaml
# The bug reproduces with any cop
Style/StringLiterals:
  Enabled: false
```

(This step is unnecessary if you use a cop that is disabled by default.)

2. Create an `example.rb` with a matching `disable` / `enable` pair:

```ruby
# frozen_string_literal: true

# rubocop:disable Style/StringLiterals
FOO = "double quoted"
# rubocop:enable Style/StringLiterals
```

3. Run `rubocop -a example.rb`

### Expected behavior

Since the `disable` and the `enable` form a matching pair, I would expect autocorrect to remove
both of them:

```ruby
# frozen_string_literal: true

FOO = "double quoted"
```

### Actual behavior

Only the `disable` is removed, and the `enable` is left behind.

Running `rubocop -a example.rb` gives:

```
$ rubocop -a example.rb
Offenses:

example.rb:3:1: W: [Corrected] Lint/RedundantCopDisableDirective: Unnecessary disabling of Style/StringLiterals.
# rubocop:disable Style/StringLiterals
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
```

`example.rb` after the correction:

```ruby
# frozen_string_literal: true

FOO = "double quoted"
# rubocop:enable Style/StringLiterals
```

Running RuboCop again does not report the leftover `enable` either:

```
$ rubocop example.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

Since #10987, `Lint/RedundantCopEnableDirective` treats an `enable` directive for a cop that is
not enabled as temporarily enabling that cop, and therefore does not register an offense. The
leftover directive is never reported, no matter how many times RuboCop runs.

As a result, autocorrect is no longer idempotent, and RuboCop considers the state it produced
itself to be clean.

For reference, when the target cop is enabled (either explicitly in `.rubocop.yml`, or by default
when left unconfigured), everything works as expected: both
`Lint/RedundantCopDisableDirective` and `Lint/RedundantCopEnableDirective` register an offense
and both directives are removed.

### RuboCop configuration

```yaml
Style/StringLiterals:
  Enabled: false
```

### RuboCop version

```shell
$ bundle exec rubocop -V
1.90.0 (using Parser 3.3.12.0, Prism 1.9.0, rubocop-ast 1.50.0, analyzing as Ruby 4.0, running on ruby 4.0.2) [arm64-darwin24]
  - rubocop-performance 1.27.0
  - rubocop-rails 2.37.0
  - rubocop-rspec 3.10.2
  - rubocop-factory_bot 2.28.0
  - rubocop-rspec_rails 2.32.0
```

**Comment by fzlzjerry** [comment]
Reproduced on current master (`d282ab04`, Ruby 3.4.10). There is a semantic distinction worth preserving from #10987: when a cop is disabled in configuration, `rubocop:enable` starts an opt-in region rather than restoring the configuration's disabled state.

For example, with the same configuration, append a statement after the final directive:

```ruby
# rubocop:disable Style/StringLiterals
FOO = "double quoted"
# rubocop:enable Style/StringLiterals
BAR = "also double quoted"
```

Running `-a` removes the first directive, leaves `FOO` unchanged, and corrects `BAR` to single quotes. Removing only the `disable` preserves that result; removing both directives stops the cop from checking `BAR`. The normal `enable` → code → `disable` opt-in case also still works.

For the original end-of-file example, two consecutive `-a` runs produce identical output, so I reproduced the leftover directive but not non-idempotent correction.

Would the intended change be limited to an `enable` with no code after it, or is this primarily a documentation gap around opt-in semantics? Removing a matching `enable` in the general case would change which code is inspected.
